cd/crossbow.py

```diff
@@ -0,0 +1,201 @@
+#!/usr/bin/env python
+
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import re
+import sys
+import click
+import pygit2
```

Comment:
This dependency is GPL (with linking exception) which is Category X for Apache projects. As this is a development only artifact, this should be no problem. We only need to take care that the script is not contained in the release (git archive) tarballs.

Reply:
BTW Later we can switch to plain git cli calls.

Reply:
Until that explicitly list the whitelist [here](https://github.com/kszucs/arrow/blob/d838c42ffb1a6f0dfa8100f0e5174c355a1d7203/dev/release/02-source.sh#L57)?

Reply:
Since this dependency isn't necessary for the operation of the Arrow libraries for normal users, it is not an issue